beetsplug/convert.py

```diff
@@ -480,6 +491,7 @@ def convert_func(self, lib, opts, args):
                 self.copy_album_art(album, dest, path_formats, pretend,
                                     link, hardlink)
 
+        self.converted_paths = set()
```

Comment:
The use of a mutable field on the plugin object creates some potential hazards for bugs in the future. Namely, it can be subtle to see how `self.converted_paths` builds up across the various function invocations. As a result, for example, it seems like it's tricky to get tests exactly right: to make sure that each "job" independently clears the set of converted paths, and that we're capturing exactly the set that we want.

There are also some potential problems with racy accesses when running conversion jobs in parallel.

Is there any chance we could instead have the set be returned from the various functions involved, such as `convert_item`? This would make it easier to see where the data flows and to keep track of the set for a given conversion "transaction."